gpt_index/indices/query/vector_store/base.py

```diff
@@ -79,6 +81,13 @@ def _retrieve(
             node_ids = [self._index_struct.nodes_dict[idx] for idx in query_result.ids]
             nodes = self._docstore.get_nodes(node_ids)
             query_result.nodes = nodes
+        else:
+            # NOTE: vector store keeps text, returns nodes.
+            # Only need to recover image or index nodes from docstore
+            for i in range(len(query_result.nodes)):
+                node_id  = query_result.nodes[i].get_doc_id()
+                if node_id in self._docstore.docs:
+                    query_result.nodes[i] = self._docstore.get_document(node_id)
```

Comment:
this basically overrides the Node object with a subclass (image or index) right?

Reply:
yes